### PR TITLE
handle columnIdx out range data size

### PR DESCRIPTION
**Pull request by bingoohuang**
1. handle columnIdx out range data size 
2. refactor code to avoid type cast

### CODE REVIEW

File: canal/sync.go [hunk 1]
```diff
@@ -245,9 +245,7 @@ func (c *Canal) updateReplicationDelay(ev *replication.BinlogEvent) {
 	atomic.StoreUint32(c.delay, newDelay)
 }
 
-func (c *Canal) handleRowsEvent(e *replication.BinlogEvent) error {
-	ev := e.Event.(*replication.RowsEvent)
-
+func (c *Canal) handleRowsEvent(e *replication.BinlogEvent, ev *replication.RowsEvent) error {
```
Comment (lance6716) [inline canal/sync.go:248]: we can change the parameter to (*EventHeader, *RowsEvent), that's not duplicated

Comment (bingoohuang) [reply canal/sync.go]: Yeah, you are right!

Comment (bingoohuang) [reply canal/sync.go]: As your suggestion, the parameters have been updated.

**Comment by skoef** [comment]
Is there perhaps a unit test for this piece of code that can be adjusted to test the specific issue @bingoohuang ran into? To prevent future regression?

**Comment by dveeden** [comment]
@bingoohuang @lance6716 what's the next step for this PR?

**Comment by lance6716** [comment]
replaced by https://github.com/go-mysql-org/go-mysql/pull/856 , because that PR has unit tests 😄

**Pull request closed without merging**